### ISSUE TITLE
New track option for "Show DTG at ends"

### ISSUE THREAD

**Issue by IanMayo** [debrief/debrief#1469]
**Note:** property now called: *Start/end time labels*

An analyst would like to always have a 6-figure DTG plotted at the start & end of a track.

Instead of Debrief setting the relevant data label to visible, the preferred behaviour is for this to be a Track setting: **Show DTG at ends**.

If set to true, when plotting a track, Debrief will show a 6 figure DTG on the first & last point.  This label will be intelligently located, to be on the Port side.

Consequently, if the user has requested a Track label to be shown, it will be placed on the Starboard side.

This functionality should not manipulate individual fixes, it should work at the top Track level.

**Comment by IanMayo** [comment]
Assigning to Russ, for documentation (probably just a small entry in the user guide).  

This feature is expected to be of value when loading tracks from a range of sources, with varying degrees of overlap.  Or, where a continuous track isn't available - just a set of disjointed sections.

**Comment by IanMayo** [comment]
Re-open. When a track has been filtered, we should show the track name at the filtered start/finish - not the first actual data point.

Being resolved in branch 1469b

**Comment by IanMayo** [comment]
Re-opened, to re-assign to Russ (for the future)

**Comment by IanMayo** [comment]
@russcrowley - I was going to discount this issue as minor, not worth taking up paper space.  But, we do aim for quite thorough coverage in the User Guide - so I think we should include it

**Comment by russcrowley** [comment]
@IanMayo - Ian, I don't think there's a specific place for this in the user guide as the UG is quite high-level where properties are concerned and to single out this for inclusion would be inconsistent with what's currently there, e.g. 'Chapter 3, Property editing' doesn't cover any individual properties. As such, I've included this in tutorial 2, in conjunction with Issue #1404 'More intelligent label location'.

**Comment by IanMayo** [comment]
@russcrowley  - yes, it's a difficult one.

I think we need to be careful to keep the pace of the tutorial high.  So, work through a breadth of different modes of Debrief usage & areas of the screen.  But, only go down into the weeds where it's necessary to keep the user engaging with the application.

So, within the above tutorials, if it would *help the plot* to Show DTG at the ends, then it's just great.  But, if it represents a *rabbit hole* then keep it out of the tutorial.

It's the User Guide that we aim to be more exhaustive.  If we don't have a placeholder for these Track Level properties then we need to consider how we support the user.

Aah,  there is something else.  Within Debrief, we do offer guidance on the meaning of Start/End Time Labels.  This is meant to be displayed in a tooltip for "mouse hover", and in the status bar when the property is selected.  Here is the guidance:

![trackwrapper](https://cloud.githubusercontent.com/assets/1108513/15315113/a2e25bf4-1c0d-11e6-9046-8365ec0bdfa6.png)

I'll create an issue to investigate that.

That leaves us with a decision.  We can't include all those properties in the Tutorial.  So, do we:
1) Choose the most significant ones, and "weave" them into the tutorial storyline
2) Choose the ones that relate to the existing storyline, and include them
3) Introduce tables of all properties (or most significant properties) in the User Guide
4) Rely on in-app tooltips to explain these properties

Or some combination of the above :-)

**Comment by russcrowley** [comment]
They all have their own merits, but it comes down to a question of urgency, resources, and direction (the last one being that to incorporate all would be a huge shift from where we're currently at).

In the meantime, functionality essential to the tutorials is essential (item 2): in its absence, there may be gaps from which increase support calls may arise. I think this is closely aligned with item 1, as the significant functions are part of the core tutorials; so that isn't, and shouldn't be, a major addition. 

Item 3 would be good at some point, but again it's about priorities and resources. 

Item 4 - definitely. It's there so we should be using it to enhance learning and help embed knowledge.

In summary, I think 1 and 2, as they're closely linked, and add tooltips where you can.

**Comment by IanMayo** [comment]
That's great @russcrowley .  I'll work on fixing 4, and we'll continue with 1 & 2.
